hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterRpcServer.java

```diff
@@ -1896,6 +1897,9 @@ public FederationRPCMetrics getRPCMetrics() {
   boolean isPathAll(final String path) {
     if (subclusterResolver instanceof MountTableResolver) {
       try {
+        if(isTrashPath(path)){
+          return true;
```

Comment:
Is that the case? Add a comment justifying.

Reply:
OK，In [HDFS-16024](https://issues.apache.org/jira/browse/HDFS-16024). The trash path is resolved by removing the prefix of the trash path. I think in the isPathAll() function, the prefix of the Trash path also needs to be processed. The isPathAll method is used to determine whether any resolved remotelocation is performed. Therefore, the isPathAll method should be consistent with the SRC path (excluding the trash prefix). Because all remotelocation resolved previously is SRC based.

Reply:
I can make two changes in isPathAll fuction：
1. Process the Trash path, remove the prefix, and check
2. Check whether it is the trash path
If we delete, mkdir, LS on the recycle bin data, if reslove gets multiple Remotelocation, then we should operate on all remotelocation, so I'm going to choose the second option. But the first is certainly fine.
If you think there's something wrong, you can talk me out of it, thank you.